### PR TITLE
Adding HerokuKafkaCliRunner, and some topic unit tests

### PR DESCRIPTION
**Pull request by noam-worthy**

### CODE REVIEW

File: src/WorthyKafkaClient.ts [hunk 1]
```diff
@@ -60,24 +98,112 @@ export class WorthyKafkaClient {
...
+        if ( KafkaOptions.useHerokuCli ) {
+            HerokuKafkaCliRunner.destroyConsumerGroup(KafkaOptions.consumer.groupId)
+        }
+    }
+
+    /**
+     * Shutdown the client. use before terminating the process.
+     */
+    public async shutdown() {
+        if ( this._consumer ) {
```
Comment (dvora-worthy) [inline src/WorthyKafkaClient.ts]: you can iterate [this._consumer, this._producer, ...]

File: src/KafkaTopicManager.ts [hunk 2]
```diff
@@ -1,22 +1,39 @@
...
         }
     }
+
+    /**
+     * Get the list of available topics
+     * @private
+     */
     private async _updateTopics() {
         // topic metadata contains all topics...
         let MD = await this._admin.fetchTopicMetadata({topics:[]})
```
[the remark this answers is not in the document]
Comment (noam-worthy) [reply src/KafkaTopicManager.ts]: yes

File: src/KafkaTopicManager.ts [hunk 3]
```diff
@@ -1,22 +1,39 @@
...
         this._admin = client.admin()
     }
 
+    /**
+     * Must run the connect method at least once before performing any topic related operation.
+     * @private
+     */
     private async _connect() {
         if ( !this._connected ) {
             await this._admin.connect()
```
[the remark this answers is not in the document]
Comment (noam-worthy) [reply src/KafkaTopicManager.ts]: of course.. thanks!

File: src/KafkaTopicManager.ts [hunk 4]
```diff
@@ -45,28 +62,87 @@ export class KafkaTopicManager {
...
+                    numPartitions: KafkaOptions.topic.partitions,
+                    replicationFactor: KafkaOptions.topic.replication
+                }], waitForLeaders: true})
+        }
+
+        await this._waitForTopic(topic)
+    }
+
+    private async _waitForTopic(topic:string,existance:boolean = true) {
+        while ( await this.topicExists(topic) !== existance ) {
```
[the remark this answers is not in the document]
Comment (noam-worthy) [reply src/KafkaTopicManager.ts]: Good point. i'll add a timeout.

File: src/WorthyKafkaClient.ts (hunk 1 above)
Comment (noam-worthy) [reply src/WorthyKafkaClient.ts]: Good idea.

**Pull request merged**